src/lib_json/json_reader.cpp

```diff
@@ -761,8 +761,8 @@ Reader::Char Reader::getNextChar() {
   return *current_++;
 }
 
-void Reader::getLocationLineAndColumn(Location location, int& line,
```

Comment:
What's wrong with int here?

https://google.github.io/styleguide/cppguide.html#Integer_Types

>  Because of historical accident, the C++ standard also uses unsigned integers to represent the size of containers - many members of the standards body believe this to be a mistake, but it is effectively impossible to fix at this point. The fact that unsigned arithmetic doesn't model the behavior of a simple integer, but is instead defined by the standard to model modular arithmetic (wrapping around on overflow/underflow), means that a significant class of bugs cannot be diagnosed by the compiler. In other cases, the defined behavior impedes optimization.

Reply:
That's a highly controversial point, and the Google style guide is not much more than an opinion piece on this topic. It's not an "accident" that C++ agrees with C on the type of its `sizeof` operator. It's also no accident that `container.size` in the STL is the same type as an array's `sizeof` operator.

In the surrounding text are listed some pitfalls of mixing arithmetic and a bit of an abdication on whether to give any advice at all w.r.t. choosing integer types, actually.

> Use care when converting integer types. Integer conversions and promotions can cause undefined behavior, leading to security bugs and other problems.

...

> That said, mixing signedness of integer types is responsible for an equally large class of problems.

The reader is left somewhat adrift. A canonical example of `size_t` causing trouble is walking a for loop backwards and underflowing at the last element. I believe it used to be in the style guide. This is not a realistically common bug, IMO. Yes, you have to know how to program in C++ to program in C++. It's not the type's fault.

For line and column numbers, I'm ok with either size_t or int. It doesn't matter much.
I'd prefer not changing it, really, as the lower-risk option.